src/sql/workbench/contrib/query/browser/messagePanel.ts

```diff
@@ -201,7 +201,7 @@ export class MessagePanel extends Disposable {
 
 	private onMessage(message: IQueryMessage | IQueryMessage[], setInput: boolean = false) {
 		if (isArray(message)) {
-			this.model.messages.push(...message);
+			this.model.messages = this.model.messages.concat(message);
```

Comment:
does this allocate and throw away an array for each message?  if so, is there any perf impact from the memory churn?

Reply:
not using concat in the new implementation anymore.